aptos-move/framework/aptos-token/sources/token_transfers.move

```diff
@@ -5,23 +5,48 @@ module aptos_token::token_transfers {
     use std::error;
     use aptos_std::table::{Self, Table};
     use aptos_token::token::{Self, Token, TokenId};
+    use aptos_framework::account;
+    use aptos_framework::event::{Self, EventHandle};
+
 
     const ETOKEN_OFFER_NOT_EXIST: u64 = 1;
 
     struct TokenTransfers has key {
         pending_claims: Table<TokenOfferId, Token>,
+        offer_event: EventHandle<TokenOfferEvent>,
+        cancel_offer_event: EventHandle<TokenCancelOfferEvent>,
+        claim_event: EventHandle<TokenClaimEvent>,
     }
 
     struct TokenOfferId has copy, drop, store {
         to_addr: address,
         token_id: TokenId,
     }
 
+    struct TokenOfferEvent has drop, store {
+        sender: address,
+        token_offer_id: TokenOfferId,
+        amount: u64,
+    }
+
+    struct TokenCancelOfferEvent has drop, store {
+        sender: address,
+        token_offer_id: TokenOfferId,
+    }
+
+    struct TokenClaimEvent has drop, store {
+        claimer: address,
+        token_offer_id: TokenOfferId,
+    }
+
```

Comment:
no need to put sender here, the event is emitted from their account
prefer we break out token offer id to be just the token_id and then have to_address at top level. It will make the devx better on the api side.
make amount consistent on each